File: Scrabble.Server/Scrabble.Server/Models/Board.cs
namespace Scrabble.Server.Models;

public class Board
{
    public Cell[][] Cells { get; }
    public int Size { get; }

    public Board(int size, Dictionary<Point, MultiplierData> multipliers)
    {
        Size = size;
        Cells = new Cell[size][];
        for (var row = 0; row < size; row++)
        {
            Cells[row] = new Cell[size];
            for (var col = 0; col < size; col++)
            {
                var point = new Point(row, col);
                var multiplier = multipliers.GetValueOrDefault(point, new MultiplierData(MultiplierType.None, 1));
                Cells[row][col] = new Cell(point, multiplier.Type, multiplier.Multiplier);
            }
        }
    }
}

File: Scrabble.Client/src/app/models/board.ts
export interface Board {
  size: number;
  cells: Cell[][];
}

export interface Cell {
  value: string | null;
  committedValue: string | null;
  multiplier?: Multiplier;
  row: number;
  col: number;
}

export type Multiplier = 'DL' | 'TL' | 'DW' | 'TW' | null;

export interface Result {
    isSuccess: boolean;
    message: string;
}

export interface TypedResult<T> {
  value?: T;
  isSuccess: boolean;
  message: string;
}

const typedResult: TypedResult<void> = {
  isSuccess: true,
  message: 'Success'
}

export interface LetterChange {
  row: number;
  col: number;
  value: string;
}
